### DISCUSSION TITLE
Sharing storage in a browser extension (chrome) between popup.js and content.js

### DISCUSSION THREAD

**Discussion by BingeCode** [joshnuss/svelte-local-storage-store#300] [Q&A]
Hi everybody,

I am currently developing a browser extension with the [wxt](https://wxt.dev/) framework and am using Svelte for the UI logic.
I wanted to refactor my code, because I am currently having to manually subscribe and unsubscribe to the cumbersome (async) local storage wrapper that is being provided by wxt.

I was looking for a cleaner solution when I stumbled across svelte-persisted-store.

I implemented it for one of my stores, but unfortunately it seems that the local storage context is not shared across the Extension Popup and the content.js scripts that get injected into the site. 

I debugged the store by employing beforeWrite and logging the values. When I changed the state in popup, I didn't get any logs in the content script and vice versa even though both were importing the store from the same file.

Am I correct in my assumption that this package is only meant for full web environments?

**Comment by joshnuss** [comment]
Hi @BingeCode,

I think it's a general question about local storage and extensions and not specific to svelte-persisted-store, which it a high-level library.

From what I can tell, this seems to be a discussion about the topic:
https://stackoverflow.com/questions/3937000/chrome-extension-accessing-localstorage-in-content-script

You may need to build something with `sendMessage` to share localStorage with the extension.
